cmd/deploy.go

```diff
@@ -472,3 +490,25 @@ func validateBuildType(buildType string) error {
 	}
 	return nil
 }
+
+func parseImageDigest(imageSplit []string, config deployConfig) error {
+
+	if !strings.HasPrefix(imageSplit[1], "sha256:") {
+		return fmt.Errorf("value '%s' in --image has invalid prefix syntax for digest (should be 'sha256:')", config.Image)
+	}
+
+	if len(imageSplit[1][7:]) != 64 {
+		return fmt.Errorf("sha256 hash in '%s' from --image has the wrong length (%d), should be 64", imageSplit[1], len(imageSplit[1][7:]))
+	}
+
+	// if BuidType was not explicitly set via CLI, set to 'disabled' without an error
+	if config.BuildType != "" && config.BuildType != "disabled" {
+		return fmt.Errorf("build type '%s' is not accepted with --image with digest. Use 'disabled' or none", config.BuildType)
```

Comment:
Nit
```suggestion
		return fmt.Errorf("build type '%s' is not accepted when using --image with digest. Use 'disabled' or none", config.BuildType)
```
Also, I worry that `use "disabled" or none` could be misleading and cause folks to use `--build none`, which will also produce an error. Here is another suggestion that could work.
```suggestion
		return fmt.Errorf("The --build flag is not valid when using --image with a digest.")
```